rust_src/src/buffers.rs

```diff
@@ -472,6 +472,119 @@ impl LispBufferLocalValueRef {
     }
 }
 
+#[no_mangle]
+pub extern "C" fn invalid_buffer_error(obj: LispObject) -> ! {
+    wrong_type!(Qbufferp, obj);
+}
+
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub enum LispBufferOrName {
+    Buffer(LispObject),
+    Name(LispObject),
+}
+
+impl LispBufferOrName {
+    pub fn as_buffer(self) -> Option<LispBufferRef> {
+        self.into()
+    }
+
+    pub fn as_buffer_or_error(self, error: extern "C" fn(LispObject) -> !) -> LispBufferRef {
```

Comment:
Which have you not seen before? Passing a function as a parameter?

Reply:
When I was looking at the code, it looked like `nsberror` what not always used. However, I have refactored the code since then. So this might be wasted code.

Reply:
I like it.